modules/gapi/test/common/gapi_imgproc_tests_common.hpp

```diff
@@ -14,6 +14,138 @@
+    cv::Mat out_mat_gapi = cv::Mat(cv::Size{ in.cols, in.rows }, in.type(), cv::Scalar::all(0));
+    cv::fillPoly(out_mat_ocv,  out_cnts_ocv,  cv::Scalar::all(1));
+    cv::fillPoly(out_mat_gapi, out_cnts_gapi, cv::Scalar::all(1));
+    EXPECT_TRUE(cmpF(out_mat_ocv, out_mat_gapi));
+}
+
+static void findContoursTestOpenCVCompare(const cv::Mat& in, const cv::RetrievalModes mode,
+                                          const cv::ContourApproximationModes method,
+                                          const std::vector<std::vector<cv::Point>>& out_cnts_gapi,
+                                          const std::vector<cv::Vec4i>&              out_hier_gapi,
+                                          const CompareMats& cmpF,
+                                          const cv::Point& offset = cv::Point())
+{
+    // OpenCV code /////////////////////////////////////////////////////////////
+    std::vector<std::vector<cv::Point>> out_cnts_ocv;
+    std::vector<cv::Vec4i>              out_hier_ocv;
+    cv::findContours(in, out_cnts_ocv, out_hier_ocv, mode, method, offset);
+    // Comparison //////////////////////////////////////////////////////////////
+    EXPECT_TRUE(out_cnts_gapi.size() == out_cnts_ocv.size());
+
+    cv::Mat out_mat_ocv  = cv::Mat(cv::Size{ in.cols, in.rows }, in.type(), cv::Scalar::all(0));
+    cv::Mat out_mat_gapi = cv::Mat(cv::Size{ in.cols, in.rows }, in.type(), cv::Scalar::all(0));
+    cv::fillPoly(out_mat_ocv,  out_cnts_ocv,  cv::Scalar::all(1));
+    cv::fillPoly(out_mat_gapi, out_cnts_gapi, cv::Scalar::all(1));
+    EXPECT_TRUE(cmpF(out_mat_ocv, out_mat_gapi));
+
+    EXPECT_TRUE(out_hier_ocv.size() == out_hier_gapi.size());
+    EXPECT_TRUE(AbsExactVector<cv::Vec4i>().to_compare_f()(out_hier_ocv, out_hier_gapi));
+}
+
+template<WithHierarchy withHierarchy = NO_HIERARCHY>
+void findContoursTestBody(const cv::Size& sz, const MatType2& type,
+                                 const cv::RetrievalModes mode,
+                                 const cv::ContourApproximationModes method,
+                                 const CompareMats& cmpF, cv::GCompileArgs&& args,
+                                 const cv::Point& offset = cv::Point())
+{
+    cv::Mat in;
+    initMatForFindingContours(in, sz, type);
+
+    std::vector<std::vector<cv::Point>> out_cnts_gapi;
+    findContoursTestGAPI(in, mode, method, std::move(args), out_cnts_gapi, offset);
+    findContoursTestOpenCVCompare(in, mode, method, out_cnts_gapi, cmpF, offset);
+}
+
+template<>
+void findContoursTestBody<WITH_HIERARCHY>(const cv::Size& sz, const MatType2& type,
+                                          const cv::RetrievalModes mode,
+                                          const cv::ContourApproximationModes method,
+                                          const CompareMats& cmpF, cv::GCompileArgs&& args,
+                                          const cv::Point& offset)
```

Comment:
Do you need a default value here?

Reply:
Here, you mean, for the `offset`? As it is the template specialization, it preserves the default values of the template; it is used here: [gapi_imgproc_tests_inl.hpp:494](https://github.com/opencv/opencv/pull/19337/files#diff-ab9e43e3581dcc571d8fe34ce51e04a8168ef8be5ae0a3eb46d3540968d31d7fR494). So, it is not necessary, it's already set.
But, I don't mind to set it here too, mb for the better understanding what's going on. Please write if you insist on that change